pkg/logql/metrics.go

```diff
@@ -133,3 +144,34 @@ func QueryType(query string) (string, error) {
 		return "", nil
 	}
 }
+
+// tagsToKeyValues converts QueryTags to form that is easy to log.
+// e.g: `Source=foo,Feature=beta` -> []interface{}{"source", "foo", "feature", "beta"}
+// so that we could log nicely!
+// If queryTags is not in canonical form then its completely ignored (e.g: `key1=value1,key2=value`)
+func tagsToKeyValues(queryTags string) []interface{} {
+	toks := strings.FieldsFunc(queryTags, func(r rune) bool {
+		return r == ','
+	})
```

Comment:
Out of curiosity, why are you not using `Split`?

Reply:
Split was a mistake (its behaviour is unintuitive), `Fields` is something usually you want to use (which was added later to stdlib. To avoid Go1 compatibility I think they put it in different name AFAIR)

Reply:
https://medium.com/@mlowicki/strings-fieldsfunc-vs-strings-split-96c667912f78

Reply:
Ah. Thanks!